### ISSUE TITLE
[BUG] SteamVR fails to use direct mode if too many screens are connected to a 7900xtx

### ISSUE THREAD

**Issue by bblacher** [ValveSoftware/SteamVR-for-Linux#612]
**Describe the bug**
My setup is as follows:
7900xtx

DP-1 3840x2160@144Hz
DP-2 1920x1080@144Hz
HDMI-A-1 1920x1080@60Hz

In order to use SteamVR in direct mode I have to use either swaymsg output DP-2 power off or swaymsg output HDMI-A-1 power off.

**To Reproduce**
Run a similar screen combination and see it fail.

**Expected behavior**
SteamVR should use direct mode

**System Information**
- Distribution: Arch
- SteamVR version: 1.27.5 beta
- Steam client version: 1695085405 
- Opted into Steam client beta?: Yes
- Graphics driver version: mesa 23.1.7
- Gist for SteamVR System Information: [gist](https://gist.githubusercontent.com/bblacher/982ed585630f27111d72db480495e796/raw/8eb90c98ae51d822926eacd2f713fb33329e8523/SteamVR%2520System%2520Info)

**Additional context**
[journal](https://gist.githubusercontent.com/bblacher/747759b3b793cb1eefb355848390bb7d/raw/5077771e16fe50ee0b5a5d98cfa95775c60ebe91/journalctl) of me trying to start SteamVR twice with all three screens connected. If I use swaymsg output xx power off to disable at least one screen SteamVR starts successfully.

**Comment by bblacher** [comment]
@Danielduel If you find some time, could you test this?

**Comment by mSparks43** [comment]
Are you sure this isn't a duplicate of #609

**Comment by bblacher** [comment]
@mSparks43 Yes, I have this issue since 1.25 and it sounds nothing like my problem.

**Comment by bblacher** [comment]
@kisak-valve Do you think the amd/drm Gitlab is the place to go for mentioning this?

**Comment by bblacher** [comment]
No I have not, I don't think it's selecting any monitor, I just think that the is some bug preventing it from entering direct mode.

Also, I don't have this issue since 1.25, I've always had this issue. It's just that 1.25 was the first version I tried. @mSparks43

**Comment by mSparks43** [comment]
> No I have not, I don't think it's selecting any monitor, I just think that the is some bug preventing it from entering direct mode.
> 
> Also, I don't have this issue since 1.25, I've always had this issue. It's just that 1.25 was the first version I tried. @mSparks43

#609 is when I switch my third monitor on it selects the wrong monitor and fails (unless I disable the monitor it is trying to select, then it uses it for the VR output instead of the headset)

**Comment by bblacher** [comment]
@mSparks43 For me it doesn't matter which monitor I disable, I just have to disable one of them.

**Comment by mSparks43** [comment]
> @mSparks43 For me it doesn't matter which monitor I disable, I just have to disable one of them.

does that disable it, or physically switch it off? different things happen for me depending on which I do.

I tracked "failing with three screens active" down to

> .local/share/Steam/logs/vrcompositor.txt

with the screen disabled it uses it instead of the headset
with the screen off it uses the headset and works fine
with the screen on, it fails because instead of

> [Info] - Found candidate direct display as RandR output 0x1ca

when it uses the headset and works fine

its picking an active display

> [Info] - Found candidate direct display as RandR output 0x1bd

and cannot acquire an xlib display that is in use.

**Comment by Danielduel** [comment]
Latest SteamVR, updated Arch
HDMI 1080p60
HDMI 1080p60
DP (1.4?) ultrawide 1440p120 (5120x1440)
Another DP - Index HMD

The problem is not present

@bblacher

**Comment by bblacher** [comment]
@Danielduel 
I do still have it, but as I said, my main screen is 2160p, so that may make the difference. Thank you for testing though :)
